app/lib/katello/util/data.rb

```diff
@@ -6,7 +6,7 @@ def self.array_with_indifferent_access(variable)
       end
 
       def self.hexdigest(string)
-        defined?(ActiveSupport::Digest) ? ActiveSupport::Digest.hexdigest(string) : Digest::MD5.hexdigest(string)
+        defined?(ActiveSupport::Digest) ? OpenSSL::Digest::SHA1.hexdigest(string)[0...32] : Digest::MD5.hexdigest(string)
```

Comment:
This forces SHA1 for hexdigest since it was default in Rails < 7. This makes sure Katello behaves the same way and the tests are green.

I'd ask maintainers to check the needed changes for this plugin to support SHA256. Afterwards this line can be switched back (until Rails upgrades the default to SHA512).

Reply:
@parthaa said in https://community.theforeman.org/t/rails-7-upgrade/37771/27 that the actual value doesn't matter, as long as it's consistent.

I wonder if the old code was ever used. https://github.com/rails/rails/commit/82822a34217503336d51b7baab82cd18cf71e435 introduced `ActiveSupport::Digest` and was introduced in Rails 5.2 (by a former Foreman developer). Perhaps a PR to replace it al with direct calls to `ActiveSupport::Digest.hexdigest()` is a better approach.

Having said that, I find it interesting that the class itself still defaults to md5: https://github.com/rails/rails/blob/4df235f7c0149ac7be86580d41a74a7785cb9bb9/activesupport/lib/active_support/digest.rb#L9

Reply:
So should I just revert the changes back and simply update IDs in VCR tests?

Reply:
Since the IDs just need to be random numbers, I think it's fine to use the new SHA 256 default. I was going to suggest continuing to use SHA1 like suggested here so the VCR recordings could be avoided, but that's already done, so I think we're all good.

Reply:
I just thought if it will break stuff for upgrading users:
 - IDs are generated or rather mapped (for the same input `A` -> the same output `B`)
 - These IDs are stored somewhere
 - Newer hashing class will create for the same input `A` -> the same output, but different `C`
 - The stored `B` can't be referenced by `A`, since hashed version is `C`.
 
 But if mapping goes only before storing and then the app using only stored values (without any further `#hexdigest` calls on `A` to reference stored value) then yeah, it should be safe. We just would need to update tests whenever new default comes from Rails.